apps/worker/src/monitor/tx.status.monitor.service.ts

```diff
@@ -85,39 +85,53 @@ export class TxStatusMonitoringService extends WorkerHost implements OnApplicati
     // do some stuff
   }
 
-  private async crawlBlockList(txHash: Hash, epoch: string, blockList: bigint[]): Promise<BlockHash | undefined> {
-    const txReceiptPromises: Promise<BlockHash | undefined>[] = blockList.map(async (blockNumber) => {
-      const blockHash = await this.blockchainService.getBlockHash(blockNumber);
-      const block = await this.blockchainService.getBlock(blockHash);
-      const txInfo = block.block.extrinsics.find((extrinsic) => extrinsic.hash.toString() === txHash.toString());
-      this.logger.debug(`Extrinsics: ${block.block.extrinsics[0]}`);
-
-      if (txInfo !== undefined) {
-        this.logger.verbose(`Found tx ${txHash} in block ${blockNumber}`);
-        const at = await this.blockchainService.api.at(blockHash.toHex());
-        const events = await at.query.system.events();
-        events.subscribe((records) => {
-          records.forEach(async (record) => {
-            const { event } = record;
-            const eventName = event.section;
-            const { method } = event;
-            const { data } = event;
-            this.logger.debug(`Received event: ${eventName} ${method} ${data}`);
-            if (eventName.search('capacity') !== -1 && method.search('Withdrawn') !== -1) {
-              const capacityWithDrawn = BigInt(data[1].toString());
-              this.logger.debug(`Capacity withdrawn: ${capacityWithDrawn}`);
-              this.setEpochCapacity(epoch, capacityWithDrawn);
-            }
-          });
-        });
-        return blockHash;
+  private async handleMessagesFailure(jobId: string, moduleError: RegistryError) {
+    this.logger.debug(`Handling extrinsic failure for job ${jobId} and module error ${moduleError}`);
+    switch (moduleError.index) {
+      // TooManyMessagesInBlock
+      case 0: {
+        break;
       }
-      return undefined;
-    });
-
-    const results = await Promise.all(txReceiptPromises);
-    const result = results.find((blockHash) => blockHash !== undefined);
-    return result;
+      // ExceedsMaxMessagePayloadSizeBytes
+      case 1: {
+        break;
+      }
+      // TypeConversionOverflow
+      // this error is returned from RPC
+      // no action needed here
+      case 2: {
+        // eslint-disable-next-line no-case-declarations
+        break;
+      }
+      // InvalidMessageSourceAccount
+      case 3: {
+        break;
+      }
+      // InvalidSchemaId
+      case 4: {
+        break;
+      }
+      // UnAuthorizedDelegate
+      case 5: {
+        break;
+      }
+      // InvalidPayloadLocation
+      case 6: {
+        break;
+      }
+      // UnsupportedCidVersion
+      case 7: {
+        break;
+      }
+      // InvalidCid
+      case 8: {
+        break;
+      }
+      default: {
+        this.logger.error(`Unknown module error ${moduleError}`);
+        break;
+      }
+    }
```

Comment:
@aramikm  give a look at these possible errors from pallet, I think we need to handle a few of them